manifests/init.pp

```diff
@@ -16,29 +16,49 @@
   $resolver_config_file_owner  = 'root',
   $resolver_config_file_group  = 'root',
   $resolver_config_file_mode   = '0644',
-) {
+  ){
 
-  # Validates domain
-  if is_domain_name($domain) != true {
-    fail("Domain name, ${domain}, is invalid.")
+  if $::kernel == 'windows' {
+    if is_array($nameservers) {
+      $stringifynameservers = join($nameservers,',')
+      $testserver = $nameservers[0]
```

Comment:
Because windows lets you specify a DNS server on every interface, for a multihoned system, I needed to figure out a way to add the DNS entries to the correct interface.  Right now I accomplish this by taking the first server and running it through find-netroute to figure out which interface would be used to get to that IP.  I then assume all the DNS servers should go on that IP.  The best scenario would probably be to evaluate each entry for it's interface and then update accordingly.  I might be able to do that in a template erb file as you recommended, or it might make sense to just write a type/provider (once I decide I want to learn that much ruby...)